### ISSUE TITLE
Component doesn't render iniitial state on error

### ISSUE THREAD

**Issue by paschaldev** [pqina/react-filepond#25]
I was debugging and I realized the components doesn't go back to it's initial state if the URL to the server returns 404. 

Here are the 3 states 

### Initial 
![image](https://user-images.githubusercontent.com/13323281/46654947-04c59c00-cba2-11e8-920f-429646936928.png)

### On upload error, the file is removed automatically 
![image](https://user-images.githubusercontent.com/13323281/46655001-2cb4ff80-cba2-11e8-9c65-e437d894a95b.png)

### Finally, it just renders a blank component. It should revert back to the initial 
![image](https://user-images.githubusercontent.com/13323281/46655021-3dfe0c00-cba2-11e8-9fa3-e259b0d9a547.png)

**Comment by rikschennink** [comment]
Can you post your `<FilePond>` set up?

**Comment by paschaldev** [comment]
```jsx 
 <FilePond 
        allowMultiple={false} 
        maxFiles={1} 
        server={config.media_upload_node}
        name="file"
        acceptedFileTypes={[
             'image/*', 
             'application/pdf', 
             'application/msword', 
             'application/vnd.openxmlformats-officedocument.wordprocessingml.document'
        ]} 
        onprocessfile={ this.fileUploadComplete }
>
        {files.map(file => (
              <File key={file} src={file} />
        ))}
                                                                        
</FilePond>
```

**Comment by rikschennink** [comment]
Is `config.media_upload_node` a URL?

What happens inside `this.fileUploadComplete`?

**Comment by paschaldev** [comment]
Yes, originally it is a URL but when I set it to an invalid URL on my server, I get the results eariler stated above.

### filUploadComplete 
```js

fileUploadComplete(error, file){

        if( error ){


        }

        const response = JSON.parse( file.serverId )

        if( response && response.media_id){

            this.setState({

                media_id: response.media_id
            })
        }
    }
```

**Comment by rikschennink** [comment]
```jsx
{files.map(file => (
              <File key={file} src={file} />
        ))}
```

Can you remove that part and see if that changes anything?

If not, can you test if removing `maxFiles={1}` changes anything, it shouldn't be necessary as `allowMultiple` is set to `false`.

**Comment by rikschennink** [comment]
Can you try updating to the latest version?

I've just updated the React component to version `5.0.0` and updated FilePond to `3.1.0`, please note that FilePond is now a peerDependency.

I have added the `onupdatefiles` callback which you can use to sync up the files list inside FilePond with the one in your component. I've extensively tested this, but still, there's a chance it might cause trouble in some situations, happy to help out if needed.

See docs for detailed information on changes:
https://pqina.nl/filepond/docs/patterns/frameworks/react/

**Comment by paschaldev** [comment]
Alright thank you @rikschennink. I'll check it out.
